### PR TITLE
Docs: Mention explicit_indirect_conditions & "Menu" -> origin_region_name

### PR DESCRIPTION
**Pull request by NewSoupVi**
https://github.com/ArchipelagoMW/Archipelago/pull/3682

### CODE REVIEW

File: docs/world api.md [hunk 1]
```diff
@@ -328,6 +328,9 @@ Even doing `state.can_reach_location` or `state.can_reach_entrance` is problemat
 You can use `multiworld.register_indirect_condition(region, entrance)` to explicitly tell the generator that, when a given region becomes accessible, it is necessary to re-check a specific entrance.
 You **must** use `multiworld.register_indirect_condition` if you perform this kind of `can_reach` from an entrance access rule, unless you have a **very** good technical understanding of the relevant code and can reason why it will never lead to problems in your case.
 
+Alternatively, you can set [world.explicit_indirect_conditions = False](https://github.com/ArchipelagoMW/Archipelago/blob/main/worlds/AutoWorld.py#L298-L301),
+in which case your world takes a flat performance hit to not have to worry about indirect conditions.
```
Comment (nicholassaylor) [inline docs/world api.md:332]:
```suggestion
Alternatively, you can set [world.explicit_indirect_conditions = False](https://github.com/ArchipelagoMW/Archipelago/blob/main/worlds/AutoWorld.py#L298-L301),
avoiding the need for indirect conditions at the expense of performance.
```
This line didn't make sense until I read the docstring that was linked to. Tried to reword to better convey that message, but I don't have any strong feelings about this change.

**Review by nicholassaylor — APPROVED** [review]
Compared the docs to the docstrings and PRs and everything matches. LGTM

**Comment by Exempt-Medic** [comment]
You'll want to also update line 469
> * additions to the regions list: at least one called "Menu"

File: docs/world api.md [hunk 2]
```diff
@@ -288,8 +288,8 @@ like entrance randomization in logic.
 
 Regions have a list called `exits`, containing `Entrance` objects representing transitions to other regions.
 
-There must be one special region, "Menu", from which the logic unfolds. AP assumes that a player will always be able to
-return to the "Menu" region by resetting the game ("Save and quit").
+There must be one special region (Called "Menu" by default, but configurable using [origin_region_name](https://github.com/ArchipelagoMW/Archipelago/blob/main/worlds/AutoWorld.py#L295-L296)),
```
Comment (Exempt-Medic) [inline docs/world api.md:291]: I worry about this link since it will have to be updated every single time anything changes in AutoWorld

Comment (NewSoupVi) [reply docs/world api.md:291]: True, not sure what else to do tho

Comment (Exempt-Medic) [reply docs/world api.md:291]: Fair enough LOL

**Review by Exempt-Medic — APPROVED** [review]
LGTM

**Review by Berserker66 — APPROVED** [review]
looks fine

**Pull request merged**